app.rb

```diff
@@ -4,123 +4,120 @@
+    when 'N'
+      permission = false
     end
+    @people << Student.new(age, nil, parent_permission: permission, name: name)
+    puts 'Student created successfully'
   end
 
-  # create rental
   def create_rental
-    puts 'Let\'s begin by selecting a book from the following list'
-    @books.each_with_index do |book, index|
-      puts "#{index}) Title: \"#{book.title}\", Author: #{book.author}"
+    unless @people.length.positive? && @books.length.positive?
+      return puts 'There should be at least one book and one person. Kindly add at least one book and one person.'
     end
-    book_index = gets.chomp.to_i
-    puts 'Now let\'s select a person from the following list'
-    @people.each_with_index do |person, index|
-      puts "#{index}) [#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+
+    puts
+    puts 'Select a book from the following list by number'
+    list_books
+    book_choice = get_input('').to_i
+    while book_choice.negative? || book_choice >= @books.length
+      book_choice = get_input("Please enter a number within 0 - #{@books.length - 1} range: ").to_i
+    end
+    book = @books[book_choice]
+    puts 'Select a person from the following list by number (not id)'
+    list_people
+    person_choice = get_input('').chomp.to_i
+    while person_choice.negative? || person_choice >= @people.length
+      person_choice = get_input("Please enter a number within 0 - #{@people.length - 1} range: ").to_i
     end
-    person_index = gets.chomp.to_i
-    print 'Lastly lets add a date:'
-    date = gets.chomp
-    rental = Rental.new(date, @books[book_index], @people[person_index])
-    @rentals.push(rental)
-    puts 'Awesome! The book has been rented'
+    person = @people[person_choice]
+    date = get_input('Enter date of booking: (yyyy/mm/dd) : ').strip
+    @rentals << person.add_rental(date, book)
   end
 
-  # List all rentals for a given person id.
-  def list_rental
-    puts 'Whose rental records would you like to see?'
+  def list_people
+    puts
     @people.each_with_index do |person, index|
-      puts "#{index}) [#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
-    end
-    person_index = gets.chomp.to_i
-    @rentals.each do |rental|
-      if rental.person.id == @people[person_index].id
-        puts "Date: #{rental.date}, Book: #{rental.book.title}, Person: #{rental.person.name}"
+      print "#{index}) [#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+      if person.instance_of?(Student)
+        puts ", Parent Permission: #{person.parent_permission}"
+      else
+        puts ", Specialization: #{person.specialization}"
       end
     end
   end
+
+  def list_rentals
+    puts
+    person_id = get_input('ID of person: ')
+    person = get_person(person_id)
+    puts 'Rentals:'
+    person.rentals.each do |rental|
+      puts "Date: #{rental.date} Book: #{rental.book.title} by #{rental.book.author}"
+    end
+  end
```

Comment:
Good job so far persisting data to a JSON file and fetching the data on app reload 👍🏽

- [ ] There is one problem currently, after adding book, person, and rental, if I try to fetch the list of rentals, the app crashes and the same happens when I reload the app with the saved data fetched, it works for the other entities but loading rentals isn't working. Kindly fix that so it loads all rentals for a person and the same for rentals persisted to the JSON file 👍🏽

<img width="1355" alt="Screenshot 2023-02-22 at 12 24 21 PM" src="https://user-images.githubusercontent.com/50941074/220620153-c1756f0b-3753-47e8-89a0-e35d16b65bb8.png">